controller/dataplane/owned_deployment.go

```diff
@@ -271,6 +275,14 @@ func listOrReduceDataPlaneDeployments(
 	return false, &deployments[0], nil
 }
 
+func podTemplateSpecHasRestartAnnotation(template *corev1.PodTemplateSpec) (bool, string) {
+	if template.Annotations == nil {
+		return false, ""
+	}
+	restartTimeStr, hasRestartAnnotation := template.Annotations[restartAnnotationKey]
+	return hasRestartAnnotation, restartTimeStr
+}
```

Comment:
Idiomatically in Go, you'd return a value first and then a boolean indicating whether it was found/valid/etc.

```suggestion
func podTemplateSpecHasRestartAnnotation(template *corev1.PodTemplateSpec) (string, bool) {
	if template.Annotations == nil {
		return "", false
	}
	return template.Annotations[restartAnnotationKey]
}
```